portalnet/src/overlay.rs

```diff
@@ -701,6 +705,26 @@ where
     pub fn get_utp_summary(&self) -> String {
         self.metrics.get_utp_summary()
     }
+
+    /// Creates an event stream channel which can be polled to receive overlay events.
+    pub fn event_stream(
+        &self,
+    ) -> impl Future<Output = anyhow::Result<mpsc::Receiver<OverlayEvent>>> + 'static {
+        let channel = self.command_tx.clone();
+
+        async move {
+            let (callback_send, callback_recv) = oneshot::channel();
+
+            let command = OverlayCommand::RequestEventStream(callback_send);
+            channel
+                .send(command)
+                .map_err(|_| anyhow!("The Overlay Service channel has been closed early."))?;
+
+            callback_recv
+                .await
+                .map_err(|_| anyhow!("The Overlay Service channel has been closed early."))
```

Comment:
*nitpick* adjust this error message so it's distinguishable from the one directly above